base/abstractarray.jl

```diff
@@ -644,8 +662,9 @@ _getindex{T,N}(::LinearSlow, A::AbstractArray{T,N}, I::Vararg{Real, N}) = (@_pro
 function _getindex(::LinearSlow, A::AbstractArray, i::Real)
     # ind2sub requires all dimensions to be > 0; may as well just check bounds
     @_inline_meta
-    @boundscheck checkbounds(A, i)
```

Comment:
That's how it was at first, but surprisingly LLVM doesn't generate very good code; there was a [10x performance regression]() despite the fact that everything is nominally inlined (these lines are the difference between the first and second benchmark reports).

I will try to dig into the underlying reasons further.